src/page/admin/auth/google-auth.jsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Grid, Cell, Switch, Textfield } from 'react-mdl';
+
+const initialState = {
+    enabled: false,
+    autoCreate: false,
+    unleashHostname: location.hostname,
+};
+
+function GoogleAuth({ config, getGoogleConfig, updateGoogleConfig, hasPermission }) {
+    const [data, setData] = useState(initialState);
+    const [info, setInfo] = useState();
+
+    useEffect(() => {
+        getGoogleConfig();
+    }, []);
+
+    useEffect(() => {
+        if (config.clientId) {
+            setData(config);
+        }
+    }, [config]);
+
+    if (!hasPermission('ADMIN')) {
```

Comment:
Not relevant for this PR, but should we extract permission types into constants?

Reply:
yes, we need to soon fix the permission systems anyways to also support project specific permissions.